src/AltinnCore/Common/Services/Implementation/DataStudioSI.cs

```diff
@@ -137,62 +141,119 @@ public object GetFormData(Guid instanceId, Type type, string applicationOwnerId,
             }
         }
 
-        /// <inheritdoc />
-        public List<AttachmentList> GetFormAttachments(string applicationOwnerId, string applicationId, int instanceOwnerId, Guid instanceId)
+        /// <inheritdoc/>
+        public async Task<List<AttachmentList>> GetFormAttachments(string applicationOwnerId, string applicationId, int instanceOwnerId, Guid instanceId)
         {
+            Instance instance;
+            List<AttachmentList> attachmentList = new List<AttachmentList>();
+            List<Attachment> attachments = null;
+            DataContractJsonSerializer serializer = new DataContractJsonSerializer(typeof(Instance));
             string developer = AuthenticationHelper.GetDeveloperUserName(_httpContextAccessor.HttpContext);
-            string attachmentsPath = $"{_settings.GetTestdataForPartyPath(applicationOwnerId, applicationId, developer)}{instanceOwnerId}/{instanceId}/data/";
-            DirectoryInfo rootDirectory = new DirectoryInfo(attachmentsPath);
-            List<AttachmentList> allAttachments = new List<AttachmentList>();
-            foreach (DirectoryInfo typeDirectory in rootDirectory.EnumerateDirectories())
+            string testDataForParty = _settings.GetTestdataForPartyPath(applicationOwnerId, applicationId, developer);
+            string formDataFilePath = $"{testDataForParty}{instanceOwnerId}/{instanceId}/{instanceId}.json";
+            string instanceData = File.ReadAllText(formDataFilePath, Encoding.UTF8);
+            instance = JsonConvert.DeserializeObject<Instance>(instanceData);
+
+            if (instance == null)
+            {
+                throw new System.NullReferenceException("Couldn't find the instance");
```

Comment:
again are we really going to throw errors?

Reply:
Fixed